### ISSUE TITLE
[Usage]: Model `compute_logits` always get None for `sampling_metadata`

### ISSUE THREAD

**Issue by yanyongyu** [vllm-project/vllm#15115]
### Your current environment

```text
vllm 0.7.3
transformers 4.49.0
```

### How would you like to use vllm

I'm writing a custom multi-modal model. The model does some special logits computes when prefill stage. I'm trying to use the `sampling_metadata` info to detect this.

It works in v0, but in v1 engine, the model always get None for `sampling_metadata` arg. How to detect prefill stage correctly?

Example code in v0:

```python
class CustomModel(nn.Module, SupportsMultiModal):
    ...
    def compute_logits(self, hidden_states: torch.Tensor, sampling_metadata: SamplingMetadata):
        logits = self.language_model.compute_logits(hidden_states, sampling_metadata)

        if any(x.seq_len is not None for x in sampling_metadata.seq_groups):
            # do something for prefill stage
```

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by yanyongyu** [comment]
It seems the metadata passed by gpu model runner is always `None`. https://github.com/vllm-project/vllm/blob/d20e261199d3e4bffe9ee0e495ba7f7454229f11/vllm/v1/worker/gpu_model_runner.py#L1048

**Comment by yanyongyu** [comment]
Could anyone answer this question please? vllm is still passing `None` to the compute_logits method in the latest code.

**Comment by WangYuzzu** [comment]
Hi, have you solved this problem?
